pkg/storage/s3/s3.go

```diff
@@ -59,7 +68,75 @@ func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
 	return nil, nil, nil
 }
 
+// checkBucketExists checks if an S3 bucket with the given name exists
+func checkBucketExists(svc *s3.S3) error {
+	_, err := svc.HeadBucket(&s3.HeadBucketInput{
+		Bucket: aws.String(d.config.Bucket),
+	})
+	return err
+}
+
+// createBucket attempts to create an s3 bucket with the given name
+func (d *driver) createBucket(svc *s3.S3) error {
+	input := &s3.CreateBucketInput{
+		Bucket: aws.String(d.Config.Bucket),
+	}
+
+	if len(d.Config.Region) > 0 && d.Config.Region != "us-east-1" {
```

Comment:
Why should the "us-east-1" region be a special case?

Reply:
It's a weird AWS API thing where us-east-1 is a special case.